### PR TITLE
Add comments to uncommented parts of code.

### PR DESCRIPTION
**Pull request by Vizvezdenec**
A lot of latest patches are not commented properly which doesn't help in understanding what parts of code actually do (mine included).
So add/clarify code in evaluate and search.
No functional change.
bench 3561701

### CODE REVIEW

**Comment by Vizvezdenec** [comment]
https://github.com/official-stockfish/Stockfish/pull/3250/files#diff-da923b7afa45cab7add143c4705b54142e46b2afe9a2627d5fa3b3474bdc8aecR1729 
       // Increase stats for the best move in case it wa a capture move
I knew I wouldn have a typo somewhere.
Also grammatical cleanup from native english speakers will be appreciated.

File: src/evaluate.cpp [hunk 1]
```diff
@@ -905,24 +908,38 @@ namespace {
...
             if (   pos.non_pawn_material(WHITE) == BishopValueMg
                 && pos.non_pawn_material(BLACK) == BishopValueMg)
                 sf = 18 + 4 * popcount(pe->passed_pawns(strongSide));
+            // For every other opposite colored bishops endgames use scale factor
+            // based on the number of all pieces of the strong side
             else
                 sf = 22 + 3 * pos.count<ALL_PIECES>(strongSide);
         }
+        // For rook endgames with strong side not having overwhelming pawn number advantage
+        // and it pawns being on one flank
```
Comment (crossbr) [inline src/evaluate.cpp:922]: and its pawns being on one flank

File: src/evaluate.cpp [hunk 2]
```diff
@@ -905,24 +908,38 @@ namespace {
...
+        // and weak side protecting it pieces with a king
```
Comment (crossbr) [inline src/evaluate.cpp:923]: and weak side protecting its pieces with a king

File: src/search.cpp [hunk 3]
```diff
@@ -1248,6 +1261,7 @@ namespace {
       {
           value = -search<NonPV>(pos, ss+1, -(alpha+1), -alpha, newDepth, !cutNode);
 
+          // If the move passed LMR update it stats
```
Comment (crossbr) [inline src/search.cpp:1264]: If the move passed LMR update its stats

File: src/search.cpp [hunk 4]
```diff
@@ -1333,6 +1346,7 @@ namespace {
           }
       }
 
+      // If move is worse than some previously searched move remember it to update it stats later
```
Comment (crossbr) [inline src/search.cpp:1349]: If move is worse than some previously searched move remember it to update its stats later

File: src/search.cpp [hunk 5]
```diff
@@ -1695,24 +1715,26 @@ namespace {
...
-        // Decrease all the non-best quiet moves
+        // Decrease stats for all non-best quiet moves
         for (int i = 0; i < quietCount; ++i)
         {
             thisThread->mainHistory[us][from_to(quietsSearched[i])] << -bonus2;
             update_continuation_histories(ss, pos.moved_piece(quietsSearched[i]), to_sq(quietsSearched[i]), -bonus2);
         }
     }
     else
+        // Increase stats for the best move in case it wa a capture move
```
Comment (crossbr) [inline src/search.cpp:1729]: Increase stats for the best move in case it was a capture move

**Comment by Vizvezdenec** [comment]
https://github.com/official-stockfish/Stockfish/pull/3250/files#diff-da923b7afa45cab7add143c4705b54142e46b2afe9a2627d5fa3b3474bdc8aecR1767
completely forgot to comment this, should be smth like
// update killer heuristic

File: src/search.cpp (hunk 4 above)
Comment (PedanticHacker) [reply src/search.cpp:1349]: Actually, a comma and a _the_ is missing, too.

Should be: _If the move is worse than some previously searched move, remember it to update its stats later_

**Comment by vondele** [comment]
@Vizvezdenec please update the patch with the review comments, so I can merge it in a next round

**Comment by Vizvezdenec** [comment]
Done.

**Comment by vondele** [comment]
thanks, also @crossbr for the review

**Pull request closed without merging**